ios/KlaviyoBridge.swift

```diff
@@ -229,6 +229,35 @@ public class KlaviyoBridge: NSObject {
             return Profile.ProfileKey.custom(customKey: str)
         }
     }
+
+    // Ensure bools and numbers are preserved as their types between the RN/Swift layers
+    static func fixNSNumberTypes(_ value: Any?) -> Any? {
+        guard let value = value, !(value is NSNull) else {
+            return nil
+        }
+
+        switch value {
+        case let num as NSNumber:
+            // Differentiate between a boolean and a number
+            if CFGetTypeID(num) == CFBooleanGetTypeID() {
```

Comment:
Pretty impressive (also seems wild that we have to do this and react native doesn't handle that correctly already but 🤷)